modules/admin/routing/routes/routes.yml

```diff
@@ -0,0 +1,24 @@
+admin-main:
```

Comment:
Is it a Symfony convention to have `routing/routes` and a `routing/services` directories? If not, I think it makes it slightly more complicated for users having their own modules. I think I would prefer to have these configuration files directly in the root directory of the module...

Reply:
We load them in different ways and I don't think we have a way to distinguish them...
[here](https://github.com/simplesamlphp/simplesamlphp/pull/1178/files/2d7943ac54b97aad6efa77d182bd738b73d89bf4#diff-35d461a964eb96cfff3e6212ba17ade7R112) and [here](https://github.com/simplesamlphp/simplesamlphp/pull/1178/files/2d7943ac54b97aad6efa77d182bd738b73d89bf4#diff-35d461a964eb96cfff3e6212ba17ade7R136)
We could lose the `routing` prefix in the path though.. Most modules will only have routes, not services I think...

Reply:
Again with the "conventions", in regards to files/structure; I would strongly suggest that we stick to the way symfony-components organizes these resources (especially for new stuff, for which backwards compability isn't an issue);
`Resources/config/services.yml`
`Resources/config/routing.yml`
Modules should also be extending Extension, and use processConfiguration (happens during building of the service container) - but I guess we need to draw the line at some point, in regards to symfonizing the application...

Reply:
Again, I don't really care for conventions that don't add anything.. We can put those files anywhere we want

Reply:
> Again, I don't really care for conventions that don't add anything.. We can put those files anywhere we want.
I agree, except with the fact that these conventions don't add anything, because they do; consistency. Which I find quite important, especially when having contributors all over the place writing code, either in core or contrib modules.
As to app/config - yes, that's where the application wide configuration is. The modules, or extensions if you may, put them in `Resources/config`.

I think the main reason for why I find reading code of Symfony projects so pleasant, is because there is consistency in where things are, and how they're named.
Not to mention the framework support from IDE's such as JetBrains - which wouldn't work (atleast not out of the box) if things are named/placed arbitrary in the codebase.

Not trying to start a flame war here, I just thought I'd point out a couple of aspects I see about this :)